### PR TITLE
test: refactor the TopologyRewriter tool

### PR DESCRIPTION
**Pull request by big-andy-coates**
#### Description 

Rework the `TopologyRewriter` tool. Better encapsulation, better structure, etc.

The tool now detects orphaned expected topology files too, and deletes them.

### Testing done 

Test only change

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksql-functional-tests/src/main/java/io/confluent/ksql/test/tools/TestExecutor.java [hunk 1]
```diff
@@ -438,25 +451,8 @@ static KsqlEngine getKsqlEngine(final ServiceContext serviceContext) {
...
-            KsqlConfig.KSQL_USE_NAMED_INTERNAL_TOPICS,
-            KsqlConfig.KSQL_USE_NAMED_INTERNAL_TOPICS_ON)
-        .put(StreamsConfig.TOPOLOGY_OPTIMIZATION, "all");
-
-    if (additionalConfigs != null) {
-      mapBuilder.putAll(additionalConfigs);
-    }
-    return mapBuilder.build();
-
+  public static Map<String, Object> baseConfig() {
```
Comment (big-andy-coates) [inline ksql-functional-tests/src/main/java/io/confluent/ksql/test/tools/TestExecutor.java]: was only ever called with an empty map

File: ksql-functional-tests/src/test/java/io/confluent/ksql/test/TopologyFileGenerator.java [hunk 2]
```diff
@@ -145,58 +140,76 @@ private static void writeExpectedTopologyFiles(
...
+        final Optional<Map<String, ?>> persistedConfigs
+    ) {
+        final KsqlConfig baseConfigs = new KsqlConfig(TestExecutor.baseConfig())
+            .cloneWithPropertyOverwrite(testCase.properties());
+
+        final KsqlConfig ksqlConfig = persistedConfigs
+            .map(baseConfigs::overrideBreakingConfigsWithOriginalValues)
+            .orElse(baseConfigs);
+
+        try (final ServiceContext serviceContext = getServiceContext();
```
Comment (purplefox) [inline ksql-functional-tests/src/test/java/io/confluent/ksql/test/TopologyFileGenerator.java]: nit: final on new line? to be consistent with our weird and ugly function parameter on new lines coding standard? ;)

File: ksql-functional-tests/src/test/java/io/confluent/ksql/test/loader/ExpectedTopologiesTestLoader.java [hunk 3]
```diff
@@ -222,7 +215,7 @@ public static TopologyAndConfigs readTopologyFile(
     final InputStream s = ExpectedTopologiesTestLoader.class.getClassLoader()
```
Comment (purplefox) [inline ksql-functional-tests/src/test/java/io/confluent/ksql/test/loader/ExpectedTopologiesTestLoader.java]: Does the stream get closed anywhere? (In a finally preferably)

File: ksql-functional-tests/src/test/java/io/confluent/ksql/test/TopologyFileGenerator.java (hunk 2 above)
Comment (big-andy-coates) [reply ksql-functional-tests/src/test/java/io/confluent/ksql/test/TopologyFileGenerator.java]: Nope, this isn't a parameter...

File: ksql-functional-tests/src/test/java/io/confluent/ksql/test/loader/ExpectedTopologiesTestLoader.java (hunk 3 above)
Comment (big-andy-coates) [reply ksql-functional-tests/src/test/java/io/confluent/ksql/test/loader/ExpectedTopologiesTestLoader.java]: Yes, by the try with resource block underneath

**Pull request merged**